internal/handlers/common/error.go

```diff
@@ -249,11 +255,47 @@ func (we *WriteErrors) Document() *types.Document {
 	return d
 }
 
+// parseArr converts err to a T type.
+// If it fails, the returned bool value is false.
+func parseErr[T any](err error) (T, bool) {
+	var obj T
+	ok := errors.As(err, &obj)
+
+	return obj, ok
+}
+
+// Append converts the err to the writeError type and
+// appends it to WriteErrors. The index value is an
+// index of the query with error.
+func (we *WriteErrors) Append(err error, index int32) {
+	if e, ok := parseErr[*writeError](err); ok {
+		e.index = &index
+		*we = append(*we, *e)
+
+		return
+	}
+
+	if e, ok := parseErr[*CommandError](err); ok {
+		*we = append(*we, writeError{err: e.Unwrap().Error(), code: e.code, index: &index})
+		return
+	}
+
+	*we = append(*we, writeError{err: err.Error(), code: errInternalError, index: &index})
+}
```

Comment:
```suggestion
func (we *WriteErrors) Append(err error, index int32) {
	var writeErr *writeError
	ok := errors.As(err, &writeErr)
	if ok {
		writeErr.index = &index
		*we = append(*we, *writeErr)

		return
	}

	var cmdErr *writeError
	ok = errors.As(err, &cmdErr)
	if ok {
		cmdErr.index = &index
		*we = append(*we, *cmdErr)

		return
	}

	*we = append(*we, writeError{err: err.Error(), code: errInternalError, index: &index})
}
```

@AlekSi Is it something you expect in place of `parseErr` function?
I've created it mainly to prevent the code repetition

Reply:
I think it will look better with a `switch`:

```go
switch {
case errors.As(err, &writeErr):
  writeErr.index = &index
  fallthrough
case errors.As(err, &cmdErr):
  cmdErr.index = &index
  *we = append(*we, err)
  return
}

My main concern with `parseError` is that it mixes regular Go interfaces and type parameters that are represented by interfaces in Go but act very differently.